LMS/library.py

```diff
@@ -0,0 +1,97 @@
+from typing import List, LiteralString, Any
+
+from book import Book
+
+
+class Library:
+    """Class Library would LIST, LEND, RETURN or ADD book(s) to the library as per the input by the user"""
+
+    def __init__(self, listOfBooks: List[Book]):
+        self.availableBooks = listOfBooks
+        self.borrowed_books = []
+
+    def displayAvailableBooks(self):
+        for book in self.availableBooks:
+            if self.is_book_available(book):
+                print(f'Book: {book.name}')
+                print(f'Author: {book.author}')
+                print(f'Number of copies available : {book.quantity}')
+                print(f'Volume: {book.volume}')
+                print("--------------")
+
+    def lendBook(self, requestedBook):
+        for book in self.availableBooks:
+            if book.name == requestedBook:
+                if self.is_book_available(book):
+                    book.quantity -= 1
+                    if self.borrowed_books:
+                        for backup_book in self.borrowed_books:
+                            if backup_book["name"] == requestedBook:
+                                backup_book["quantity_borrowed"] += 1
+                    else:
+                        self.borrowed_books.append(
+                            {"name": requestedBook,
+                             "quantity_borrowed": 1}
+                        )
+                    print(f"Great choice! {requestedBook} is now yours. Kindly return it within next 2 weeks!!")
+                    return
+        else:
+            print("Sorry, the book you requested isn't currently available in the library")
+
+    def addBook(self, newBook, qty):
+        author = input("Please enter the author of the book: ")
+        vol = input("Please enter the volume of the book: ")
+        self.availableBooks.append(Book(newBook, author, qty, vol))
+        print("Thanks for the new addition to our Library!")
+
+    def returnBook(self, returnedBook):
+        qty = int(input("How many copies would you like to return? "))
+        check = False
+        if self.borrowed_books:
+            for borrowed_book in self.borrowed_books:
+                if borrowed_book["name"] == returnedBook:
```

Comment:
This is probably a logical flaw, book is not uniquely identifiable by its name. Also, comparison of two books should always be done the methods we defined in book class